runtime/runtime-core/jvm/src/aws/smithy/kotlin/runtime/content/ByteStreamJVM.kt

```diff
@@ -45,7 +45,14 @@ public fun Path.asByteStream(start: Long = 0, endInclusive: Long = -1): ByteStre
     val f = toFile()
     require(f.exists()) { "cannot create ByteStream, file does not exist: $this" }
     require(f.isFile) { "cannot create a ByteStream from a directory: $this" }
-    return f.asByteStream(start, endInclusive)
+
+    val calculatedEndInclusive = when {
+        endInclusive >= 0L -> endInclusive
+        endInclusive == -1L -> f.length() - 1
+        else -> throw IllegalArgumentException("endInclusive must be -1 or greater")
```

Comment:
style/nit: `error(...)`

Reply:
err yes `require` I guess is what I meant, this isn't that big of deal you can leave it as is